server/gendb/interface.go.tmpl

```diff
@@ -136,6 +153,18 @@ type Get{{ $modelName }}sBy{{ $indexName }}Input struct {
   DisableConsistentRead bool
   // Limit is an optional limit of how many items to evaluate.
 	Limit  *int64
+  {{- if gt $lenNonIndexAttributes 0 }}
+    // Filters is an optional array of filters to apply on various table attributes
+    FilterValues []{{ $modelName }}By{{ $indexName }}FilterValues
+    // FilterExpression is the filter expression to be applied to our fitlered attributes 
+    // when referencing an attribute use #ATTRIBUTE_NAME
+    // ex: if the attribute is called "created_at" in its wag definition use #CREATED_AT
+    // when referencing one of the given values use :{attribute_name}_value0, :{attribute_name}_value1, etc.
+    // ex: if the attribute is called "created_at" in its wag definition use :created_at_value0, created_at_value1, etc.
+    // see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.html#Query.KeyConditionExpressions
+    // for guidance on building expressions
+    FilterExpression  string
```

Comment:
Is there any validation to ensure that if FilterValues are provided that a FilterExpression is also provided? If not, what would it mean if only one was provided

Reply:
No validation that throws an error but we have this at line 557 `if len(input.FilterValues) > 0 && input.FilterExpression != ""`. if both are not provided then no filters will be applied

Reply:
Could you add a comment above that line explaining why we are doing that check? Just so it's clear that both fields are needed to filter, and if one is missing the other is ignored